schema.c

```diff
@@ -411,6 +412,22 @@ add_module_info_to_children (xmlNode *node, xmlNsPtr ns, xmlChar *mod, xmlChar *
                 xmlNewProp (n, (const xmlChar *)"version", ver);
                 xmlNewProp (n, (const xmlChar *)"features", feat);
                 xmlNewProp (n, (const xmlChar *)"deviations", devi);
+                s = sch_node_next_sibling ((sch_node *) n);
```

Comment:
this appears to be the same loop that is being processed at the top level with the n= n->next

the only difference appears to be the check for the node being an XML_ELEMENT_NODE with name "NODE"

does that mean we are missing that check  and need the check above?